File: sdk/src/main/java/com/meniga/sdk/webservices/requests/CreateBudgetEntry.java
package com.meniga.sdk.webservices.requests;

import com.meniga.sdk.helpers.MenigaDecimal;

import org.joda.time.DateTime;

import java.util.List;

/**
 * Copyright 2017 Meniga Iceland Inc.
 */

public class CreateBudgetEntry extends QueryRequestObject {

	public MenigaDecimal targetAmount;
	public DateTime startDate;
	public DateTime endDate;
	public List<Long> categoryIds;
	public int autoFillType;

	@Override
	public long getValueHash() {
		return 0;
	}
}

File: sdk/src/main/java/com/meniga/sdk/webservices/budget/CreateBudgetEntry.kt
/*
 * Copyright 2017 Meniga Iceland Inc.
 */
package com.meniga.sdk.webservices.budget

import com.meniga.sdk.helpers.MenigaDecimal
import com.meniga.sdk.webservices.requests.QueryRequestObject

import org.joda.time.DateTime

internal data class CreateBudgetEntries(val entries: List<CreateBudgetEntry>) : QueryRequestObject() {
    override fun getValueHash(): Long = hashCode().toLong()
}

internal data class CreateBudgetEntry(
        var targetAmount: MenigaDecimal? = null,
        var startDate: DateTime? = null,
        var endDate: DateTime? = null,
        var categoryIds: List<Long>? = null
)
